src/plugins/visualize/public/application/utils/get_table_columns.tsx

```diff
@@ -87,20 +87,24 @@ export const getTableColumns = (application: ApplicationStart, history: History)
       defaultMessage: 'Title',
     }),
     sortable: true,
-    render: (field: string, { editApp, editUrl, title }: VisualizationListItem) => (
-      <EuiLink
-        onClick={() => {
-          if (editApp) {
-            application.navigateToApp(editApp, { path: editUrl });
-          } else if (editUrl) {
-            history.push(editUrl);
-          }
-        }}
-        data-test-subj={`visListingTitleLink-${title.split(' ').join('-')}`}
-      >
-        {field}
-      </EuiLink>
-    ),
+    render: (field: string, { editApp, editUrl, title, error }: VisualizationListItem) =>
+      // In case an error occurs i.e. the vis has wrong type, we render the vis but without the link
+      !error ? (
+        <EuiLink
+          onClick={() => {
+            if (editApp) {
+              application.navigateToApp(editApp, { path: editUrl });
+            } else if (editUrl) {
+              history.push(editUrl);
+            }
+          }}
+          data-test-subj={`visListingTitleLink-${title.split(' ').join('-')}`}
+        >
+          {field}
+        </EuiLink>
+      ) : (
+        field
+      ),
   },
```

Comment:
I would render a string stating "Unknown Type" into the type column, to clarify what happened, together with a "warning" icon.

Might even consider adding a tooltip with more details.

Other than that looks good.